### ISSUE TITLE
Overrides do not work with extends/package.json

### ISSUE THREAD

**Issue by milesj** [eslint/eslint#9025]
**Tell us about your environment**

* **ESLint Version:** 4.3
* **Node Version:** 8.2.1
* **npm Version:** 5.3.0
* **Yarn version:** 0.27.5

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint

**Please show your full configuration:**

https://github.com/milesj/build-tool-config/blob/master/eslint.node.json5

**What did you do? Please include the actual source code causing the issue.**

Overrides are not respected when they exist in a configuration file that is extended from.

```js
overrides: [
    {
      files: ['tests/**/*.test.js'],
      plugins: ['jest'],
      env: {
        jest: true,
      },
      rules: {
        'no-magic-numbers': 'off',
        'jest/no-identical-title': 'error',
        'jest/valid-expect': 'error',
      },
    },
],
```

**What did you expect to happen?**

Overrides work.

**What actually happened? Please include the actual, raw output from ESLint.**

None of the settings or rules in the overrides block are inherited. I believe the issue is that override globs are relative to the configuration file path, instead of the current working directory.

If I extend from a config file, or from package.json, `overrides` are broken. For example: https://github.com/milesj/interweave/blob/master/package.json#L68

**Comment by not-an-aardvark** [comment]
Closing as a duplicate of https://github.com/eslint/eslint/issues/8813

**Comment by milesj** [comment]
Cheers!
